pkg/beholder/node_auth.go

```diff
@@ -0,0 +1,83 @@
+package beholder
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+)
+
+// NodeAuthenticator implements PerRPCCredentials interface from google.golang.org/grpc/credentials
+type NodeAuthenticator struct {
+	staticAuthHeaders        map[string]string
+	requireTransportSecurity bool
+}
+
+func NewNodeAuthenticator(config Config) (*NodeAuthenticator, error) {
```

Comment:
Does this need to be exported?
```suggestion
// nodeAuthenticator implements PerRPCCredentials interface from google.golang.org/grpc/credentials
type nodeAuthenticator struct {
	staticAuthHeaders        map[string]string
	requireTransportSecurity bool
}

func newNodeAuthenticator(config Config) (*NodeAuthenticator, error) {
```